src/Elrond/GasEstimator.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "GasEstimator.h"
+
+#include "../proto/Elrond.pb.h"
+
+using namespace TW;
+using namespace TW::Elrond;
+
+GasEstimator::GasEstimator(const NetworkConfig& networkConfig) {
+    this->networkConfig = networkConfig;
+}
+
+long GasEstimator::forEGLDTransfer(long dataLength) {
+    long gasLimit = 
+        this->networkConfig.getMinGasLimit() + 
+        this->networkConfig.getGasPerDataByte() * dataLength;
+
+    return gasLimit;
+}
+
+long GasEstimator::forESDTTransfer(long dataLength) {
+    // Additional gas to account for eventual increases in gas requirements (thus avoid breaking changes in clients of TW-core).
+    const long ADDITIONAL_GAS = 100000;
```

Comment:
can the constants be moved outside the functions?

Reply:
Moved.